Reading List/Reading List/Models/BookController.swift

```diff
@@ -0,0 +1,103 @@
+//
+//  BookController.swift
+//  Reading List
+//
+//  Created by Ian Becker on 5/19/20.
+//  Copyright © 2020 Lambda School. All rights reserved.
+//
+
+import Foundation
+
+class BookController {
+    var books: [Book] = []
+    var readBooks: [Book] {
+        books.filter {
+            $0.hasBeenRead == true
+        }
+    }
+    
+    var unreadBooks: [Book] {
+        books.filter {
+            $0.hasBeenRead == false
+        }
+    }
+    
+    
+    init() {
+        loadFromPersistentStore()
+    }
+    
+    func createBook(title: String, reasonToRead: String, hasBeenRead: Bool) {
+        let book = Book(title: title, reasonToRead: reasonToRead, hasBeenRead: hasBeenRead)
+        
+        books.append(book)
+        saveToPersistentStore()
+    }
+    
+    func deleteBook(book : Book) {
+        guard let index = books.firstIndex(of: book) else { return }
+        books.remove(at: index)
+        saveToPersistentStore()
+    }
+    
+    func updateHasBeenRead(for book: Book) {
+        guard let index = books.firstIndex(of: book) else { return }
+        
+        books[index].hasBeenRead.toggle()
+        saveToPersistentStore()
+        }
+    
+    func editBook(for book: Book) {
```

Comment:
This code is a form of recursion. Your function declaration should have more arguments to pass in.